build_library/vm_image_util.sh

```diff
@@ -1199,45 +1199,54 @@ vm_cleanup() {
 }
 
 vm_upload() {
-    local digests="$(_dst_dir)/$(_dst_name .DIGESTS)"
-    upload_image -d "${digests}" "${VM_GENERATED_FILES[@]}"
-
-    [[ -e "${digests}" ]] || return 0
 
-    # FIXME(marineam): Temporary alternate name for .DIGESTS
-    # This used to be derived from the first file listed in
-    # ${VM_GENERATED_FILES[@]}", usually $VM_DST_IMG or similar.
-    # Since not everything actually uploads $VM_DST_IMG this was not very
-    # consistent and relying on ordering was breakable.
-    # Now the common prefix, output by $(_dst_name) is used above.
-    # Some download/install scripts may still refer to the old name.
-    local uploaded legacy_uploaded
-    for uploaded in "${VM_GENERATED_FILES[@]}"; do
-        if [[ "${uploaded}" == "${VM_DST_IMG}" ]]; then
-            legacy_uploaded="$(_dst_dir)/$(basename ${VM_DST_IMG})"
-            break
+    declare -a legacy_uploads
+    declare -a uploadable_files
+    declare -a compressed_images
+    declare -a image_files
+    declare -a digest_uploads
+
+    for file in "${VM_GENERATED_FILES[@]}";do
+        if [[ "${file}" =~ \.(img|bin|vdi|vhd|vmdk)$ ]]; then
```

Comment:
This duplicates the logic in `compress_disk_images` and I think it's better to move it there completely as suggested above

Reply:
Ok, this legacy logic is a bit complicated, thanks for your works there. I think it's good now, or?

Reply:
Yup. Should be ok. The changes to ```compress_disk_image()``` makes is also populate an "extra_files" array with whatever doesn't match that image pattern. Which callers can then concatenate and upload if they wish.